### PR TITLE
Tasks from 1 to 5.

### PR DESCRIPTION
**Pull request by martinmilev**
First 5 tasks from the theme.

### CODE REVIEW

**Comment by mgenov** [comment]
You are not using the correct .gitignore and I'm going to review IDEA project files !

File: src/ex2/Writer.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    writing(exitSymbol);
+  }
+
+  /**
+   * Reading from the console input until a new line wint ".".
+   *
+   * @return
+   * @param exitSymbol
+   */
+  private String reading(String exitSymbol) {
```
Comment (mgenov) [inline src/ex2/Writer.java]: reading is not verb.

File: src/ex3/Reverser.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ */
+public class Reverser {
+
+  /**
+   * Reversing the content of a text file.
+   *
+   * @param filename
+   * @throws IOException
+   */
+  public void reversing(String filename) throws IOException {
```
Comment (mgenov) [inline src/ex3/Reverser.java]: reversing is not a verb.

File: src/ex3/Reverser.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by clouway on 20.04.16.
+ */
+public class Reverser {
```
Comment (mgenov) [inline src/ex3/Reverser.java]: I think that this class is duplicated with TextFile, so one of them need to be removed.

File: src/ex4/TransferObject.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+public class TransferObject {
+
+  /**
+   * Transfer data from one stream to another.
+   * Transfering from file to file.
+   * @param filein
+   * @param fileout
+   * @throws IOException
+   */
+  public void transfer(String filein, String fileout) throws IOException {
```
Comment (mgenov) [inline src/ex4/TransferObject.java]: You need to be provide offset and number of bytes to be transferred. 

This is what task says: "Да се обмислят и проверят възможностите за частично прехвърляне."

File: src/ex5/Demo.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Created by clouway on 21.04.16.
+ */
+public class Demo {
+  public static void main(String[] args) {
+    List<String> list = new LinkedList<>();
+    DirectoryBrowser db1 = new DirectoryBrowser();
+    DirectoryBrowser db2 = new DirectoryBrowser();
+    db1.listContent(list, "/home/clouway/workspace/idea/IO/file1");
```
Comment (mgenov) [inline src/ex5/Demo.java]: I don't have this directory on my pc which means that this demo won't work on my side. It's a bad idea for using such path. Maybe some system path or current folder will be much better.

File: src/ex5/DirectoryBrowser.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+   * @return
+   */
+  public List<String> listContent(List<String> list, String path) {
+    Path place = Paths.get(path);
+    File file = new File(path);
+
+    if (file.exists()) {
+      if (file.isFile()) {
+        list.add("File");
+        list.add("Name: " + file.getName());
```
Comment (mgenov) [inline src/ex5/DirectoryBrowser.java]: Printing is still in here where it should be in the Demo.

Comment (martinmilev) [inline src/ex5/DirectoryBrowser.java]: It's printing the List<String> in the demo.
I could try printing it some other way if it is needed.

Comment (mgenov) [inline src/ex5/DirectoryBrowser.java]: I see, you are putting things in list and then you are printing the list.

How can I list all files in a folder and print these with length bigger then 100 bytes? 

```java
List<String> files = directoryBrowser.listContent(new ArrayList<String,String>(), "/home/clouway/temp");
```

File: src/ex5/DirectoryBrowser.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+public class DirectoryBrowser {
+
+  /**
+   * Listing content at file path.
+   *
+   * @param list
+   * @param path
+   * @return
+   */
+  public List<String> listContent(List<String> list, String path) {
```
Comment (mgenov) [inline src/ex5/DirectoryBrowser.java]: Passing of list seems use-less in the context of single invocation. I saw that in demo, you are passing single list to multiple invocations, but it doesn't look reasonable. 

You can do the following instead: 

```java
List<String> content1 = browser.listContent("/dir1");
List<String> content2 = browser.listContent("/dir2"); 
Iterable<String> allFoldersContent = Iterables.concat(content1, content2);

for (String each : allFoldersContent) {
   System.out.println(each);
}
```

So the caller will decide whether objects should be combined or not. Passing of params that mutate during the call is not a good idea.

**Pull request closed without merging**